R/misc.R

```diff
@@ -0,0 +1,181 @@
+#' Create a Document Position
+#'
+#' Creates a \code{document_position}, which can be used to indicate e.g. the
+#' row + column location of the cursor in a document.
+#'
+#' @param x An object coercable to \code{document_position}.
+#' @param row The row (using 1-based indexing).
+#' @param column The column (using 1-based indexing).
+#'
+#' @name document_position
+#'
+#' @export
+#' @family location
+makePosition <- function(row, column) {
+  structure(c(row = as.numeric(row), column = as.numeric(column)),
+            class = "document_position")
+}
+
+#' @rdname document_position
+#' @export
+is.document_position <- function(x) {
+  inherits(x, "document_position")
+}
+
+#' @name document_position
+#' @export
+as.document_position <- function(x) {
```

Comment:
I think it's a bit better to make this generic and have both `as.document_position.numeric()` and `as.document_position.document_position()`.